server/src/main/scala/za/co/absa/atum/server/api/service/DatabaseService.scala

```diff
@@ -38,4 +37,13 @@ class DatabaseService @Autowired()() {
     } yield checkpoint
   }
 
+  /**
+   * Function to retrieve checkpoint based on the provided fields
+   *
+   * @param filterCriteria JSON object containing the fields for filtering the checkpoint
+   */
+  def createPartitioningIfNotExists(partitioningInfo: PartitioningDTO): Seq[PartitionDTO] = {
```

Comment:
I think that the function you are trying to call here is not gonna return PartitionDTO sequence, but something else, check the DB function definition:

```
CREATE OR REPLACE FUNCTION runs.create_partitioning_if_not_exists(
    IN  i_partitioning          JSONB,
    IN  i_by_user               TEXT,
    IN  i_parent_partitioning   JSONB = NULL,
    OUT status                  INTEGER,
    OUT status_text             TEXT,
    OUT id_partitioning         BIGINT
)
```